src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -516,7 +517,7 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
   public async executeQuery() {
     this.showWaitingAnimation();
     try {
-      await this.queryController.executeQuery();
+      await this.queryController.executeQuery({ closeModalBox: !ResponsiveComponentsUtils.isSmallFacetActivated($$(this.root)) });
```

Comment:
There's some duplication where the `closeModalBox` statement is being repeated in the different facets.

Would it be possible for the ModalBox to listen for the `executeQuery`, and close itself if the condition is met?

This would reduce the chance of an issue if someone creates a custom facet, and forgets to set this property when making a query.

Reply:
Well, the thing is sometimes you want to close it, other time you want to leave it open. It depends on the context, and where the executeQuery function is called. 

Hence the control being at the caller location, not the modal box itself.